packages/sign_in_with_apple/android/src/main/kotlin/com/aboutyou/dart_packages/sign_in_with_apple/SignInWithApplePlugin.kt

```diff
@@ -0,0 +1,131 @@
+import android.os.Bundle;
+import androidx.annotation.NonNull;
+import io.flutter.embedding.engine.plugins.FlutterPlugin
+import io.flutter.plugin.common.MethodCall
+import io.flutter.plugin.common.MethodChannel
+import io.flutter.plugin.common.MethodChannel.MethodCallHandler
+import io.flutter.plugin.common.MethodChannel.Result
+import io.flutter.plugin.common.PluginRegistry.Registrar
+
+import android.net.Uri
+import android.content.Intent
+import androidx.browser.customtabs.CustomTabsIntent
+
+import io.flutter.embedding.engine.plugins.activity.ActivityAware
+import io.flutter.embedding.engine.plugins.activity.ActivityPluginBinding
+
+/** SignInWithApplePlugin */
+public class SignInWithApplePlugin: FlutterPlugin, MethodCallHandler, ActivityAware {
+  private var channel: MethodChannel? = null
+
+  var activity: Activity? = null
+  
+
+  override fun onAttachedToEngine(@NonNull flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
+    channel = MethodChannel(flutterPluginBinding.binaryMessenger, "com.aboutyou.dart_packages.sign_in_with_apple")
+    channel?.setMethodCallHandler(this);
+  }
+
+  override fun onDetachedFromEngine(@NonNull binding: FlutterPlugin.FlutterPluginBinding) {
+    channel?.setMethodCallHandler(null)
+    channel = null
+  }
+
+  // This static function is optional and equivalent to onAttachedToEngine. It supports the old
+  // pre-Flutter-1.12 Android projects. You are encouraged to continue supporting
+  // plugin registration via this function while apps migrate to use the new Android APIs
+  // post-flutter-1.12 via https://flutter.dev/go/android-project-migration.
+  //
+  // It is encouraged to share logic between onAttachedToEngine and registerWith to keep
+  // them functionally equivalent. Only one of onAttachedToEngine or registerWith will be called
+  // depending on the user's project. onAttachedToEngine or registerWith must both be defined
+  // in the same class.
+  companion object {
+    var lastAuthorizationRequestResult: Result? = null
+    var triggerMainActivityToHideChromeCustomTab : (() -> Unit)? = null
+
+    @JvmStatic
+    fun registerWith(registrar: Registrar) {
+      val channel = MethodChannel(registrar.messenger(), "com.aboutyou.dart_packages.sign_in_with_apple")
+      channel.setMethodCallHandler(SignInWithApplePlugin())
+    }
+  }
+
+  override fun onMethodCall(@NonNull call: MethodCall, @NonNull result: Result) {
+    when (call.method) {
+      "isAvailable" -> result.success(true)
+      "performAuthorizationRequest" -> {
+        val _activity = activity
+
+        if (_activity == null) {
+          result.error("MISSING_ACTIVITY", "Plugin is not attached to an activity", call.arguments)
+          return
+        }
+
+        val url: String? = call.argument("url")
+
+        if (url == null) {
+          result.error("MISSING_ARG", "Missing 'url' argument", call.arguments)
+          return
+        }
+
+        SignInWithApplePlugin.lastAuthorizationRequestResult = result
```

Comment:
@tp I think we should check if we already have an existing `lastAuthorizationRequestResult` here and in case, we should complete that one with an error

Reply:
Makes sense, as this can happen when the Custom Tab gets closes by the user.

Will look if this can be detected and resolve the pending task then as well, as that'd be preferable.